### PR TITLE
:sparkles: [open-formulieren/open-forms#4199] Explicitly start anonymous submission

### PR DESCRIPTION
**Pull request by stevenbal**
Issue: https://github.com/open-formulieren/open-forms/issues/4199

Related PR in backend: https://github.com/open-formulieren/open-forms/pull/4271

### CODE REVIEW

File: src/components/Form.js [hunk 1]
```diff
@@ -178,7 +178,14 @@ const Form = () => {
 
     let submission;
     try {
-      submission = await createSubmission(config.baseUrl, form, config.clientBaseUrl);
+      const anonymous = !!event;
```
Comment (stevenbal) [inline src/components/Form.js:181]: I'm not sure if there is another way for me to figure out what login option the user selected at this point

Comment (sergei-maertens) [reply src/components/Form.js:181]: yeah, in `src/components/LoginOptions/index.js` on line 66 you can provide an explicit argument to indicate that it's an anonymous login:

```jsx
  const containerProps = form.loginRequired
    ? {}
    : {
        onSubmit: e => {
          e.preventDefault();
          onFormStart(e, {anonymous: true});
        },
      };
```
so then just update the `onFormStart` signature to take `(event, {anonymous = false}) {...}`, that makes it explicit :)

File: src/components/Form.js [hunk 2]
```diff
@@ -178,7 +178,13 @@ const Form = () => {
 
     let submission;
     try {
-      submission = await createSubmission(config.baseUrl, form, config.clientBaseUrl);
+      submission = await createSubmission(
```
Comment (stevenbal) [inline src/components/Form.js:181]: @sergei-maertens I don't think this line was covered before either, and I'm not sure where to start with writing a test for this `onFormStart` :sweat_smile:

**Review by sergei-maertens — APPROVED** [review]
It was painful, but I managed a test for the non-anonymous flow too.

**Pull request merged**